### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by nmclear**
Up to date code as of 6/27 11:30am.

### CODE REVIEW

File: models/inventory.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+    return Inventory;
+}
+
+
+
+
+
+// store_id: {
```
Comment (danieldealday) [inline models/inventory.js]: Comments on page can be left during development, however, be sure to leave notes indicating that it is to saved their for future use. Be sure to annotate your commented code.

File: routes/inventory-routes.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    });
+
+    // GET inventory by seller
+    app.get("/api/inventory/seller/:id", function(req,res){
+
+        db.Inventory.findAll({
+            where: {
+                sellerId: req.params.id
+            },
+            include: [db.Products][db.Seller]
```
Comment (jtblanche) [inline routes/inventory-routes.js]: does this work?  if it doesn't maybe try [db.Products, db.Seller], if it does work, let us know how.

**Review by jtblanche — CHANGES_REQUESTED** [review]
my first set of comments, I may make more at some point ;)

File: config/config.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require('dotenv').config();
```
Comment (ALMaclaine) [inline config/config.js]: Why do you have a config.js and config json that are nearly identical.

File: routes/login-routes.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+const db = require("../models");
+
+module.exports = function(app){
+
+
+    // POST new user
+    app.post("/api/users", function(req, res){
+        db.Products.create(req.body).then(function(dbUser){
```
Comment (ALMaclaine) [inline routes/login-routes.js]: Avoid passing in the whole body incase it's api changes in the future, always grab off exactly what you need.
Use es6 object initialization like I showed in the other comments.

{
  req.body.property1,
  req.body.property2,
  req.body.property3
}

File: routes/login-routes.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        where: {
+            id: req.params.id
+        }
+        }).then(function(dbUser) {
+        res.json(dbUser);
+        });
+    });
+
+    // UPDATE user info
+    app.put("/api/user", function(req, res) {
```
Comment (ALMaclaine) [inline routes/login-routes.js]: Avoid passing in the whole body incase it's api changes in the future, always grab off exactly what you need.
Use es6 object initialization like I showed in the other comments.

{
  req.body.property1,
  req.body.property2,
  req.body.property3
}

File: routes/products-routes.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        }).then(function(dbProduct) {
+        res.json(dbProduct);
+        });
+    });
+
+
+
+    // POST new products (may not need)
+    app.post("/api/products", function(req, res){
+        db.Products.create(req.body).then(function(dbProducts){
```
Comment (ALMaclaine) [inline routes/products-routes.js]: Avoid passing in the whole body incase it's api changes in the future, always grab off exactly what you need.
Use es6 object initialization like I showed in the other comments.

{
  req.body.property1,
  req.body.property2,
  req.body.property3
}

File: routes/seller-routes.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            }
+            }).then(function(dbSellers) {
+            res.json(dbSellers);
+        });
+    });
+
+    // POST new seller
+    app.post("/api/sellers", function(req,res){
+        // may not be req.body. Check after AJAX Setup. Need to create object.
+        db.Seller.create(req.body).then(function(dbSellers){
```
Comment (ALMaclaine) [inline routes/seller-routes.js]: Avoid passing in the whole body incase it's api changes in the future, always grab off exactly what you need.
Use es6 object initialization like I showed in the other comments.

{
  req.body.property1,
  req.body.property2,
  req.body.property3
}

File: public/assets/javascript/app.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+$(document).ready(function(){
```
Comment (ALMaclaine) [inline public/assets/javascript/app.js]: prefer arrows for simple functions, especially if 'this' is not involved.
$(document).ready(() => $('.parallax').parallax())

File: public/assets/js/product-public.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+
+    function selectProduct(event){
+        event.preventDefault();
+
+        // wherever we are selecting products we need to store the product ID as val or some datatype.
+        // products should NOT be input. Products should be list.
+        const productId = $('.selectProduct').val().trim();
+
+        if(!productId) {
```
Comment (ALMaclaine) [inline public/assets/js/product-public.js]: if(productId) {
 console.log(`Product ID ${ProductID}`) 
}

File: public/assets/js/seller-public.js [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+                // add to a div to show all sellers
+            }
+        });
+    };
+
+
+    // View seller by id
+    function viewSellerById(id){
+
+        if(!NaN(id)){
```
Comment (ALMaclaine) [inline public/assets/js/seller-public.js]: Don't believe NaN is a function.

File: routes/inventory-routes.js (hunk 2 above)
Comment (ALMaclaine) [inline routes/inventory-routes.js]: This doesn't appear to be correct, please explain this to one of us if you find it working.

File: routes/inventory-routes.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+        }).then(function(dbInventory){
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: Prefer arrow functions for short functions without this.
.then((dbInventory) => res.json(dbInventory))

File: routes/inventory-routes.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    });
+
+    // GET inventory by product
+    app.get("/api/inventory/product/:id", function(req,res){
+
+        db.Inventory.findAll({
+            where: {
+                productId: req.params.id
+            },
+            include: [db.Products][db.Seller]
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: This doesn't appear to be correct, please explain this to one of us if you find it working.

File: routes/inventory-routes.js [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    // POST new inventory
+    app.post("/api/inventory", function(req, res) {
+        const newInventory = req.body;
+
+        db.Inventory.create({
+            stock: newInventory.stock,
+            productId: newInventory.productId,
+            sellerId: newInventory.sellerID
+
+        }).then(function(dbInventory) {
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: Prefer arrow functions for short functions without this.
.then((dbInventory) => res.json(dbInventory))

File: routes/inventory-routes.js [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+            res.json(dbInventory);
+        });
+    });
+
+
+    // POST new inventory
+    app.post("/api/inventory", function(req, res) {
+        const newInventory = req.body;
+
+        db.Inventory.create({
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: Prefer es6 initializer syntax
{
stock,
productId,
sellerId
}

File: routes/inventory-routes.js [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+    app.put("/api/inventory/", function(req, res) {
+    
+            db.Inventory.update(
+                req.body,
+                {
+                    where: {
+                        //this should be inventory unique ID
+                        id: req.body.id
+                    }  
+            }).then(function(dbInventory) {
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: Prefer arrow functions for short functions without this.
.then((dbInventory) => res.json(dbInventory))

File: routes/inventory-routes.js [hunk 16]
```diff
@@ -0,0 +1,85 @@
...
+
+
+    // DELETE inventory
+    app.delete("/api/inventory/:id", function(req,res){
+
+        db.Inventory.destroy({
+            where: {
+                id: req.params.id
+            },
+        }).then(function(dbInventory){
```
Comment (ALMaclaine) [inline routes/inventory-routes.js]: Prefer arrow functions for short functions without this.
.then((dbInventory) => res.json(dbInventory))

File: routes/login-routes.js [hunk 17]
```diff
@@ -0,0 +1,37 @@
+const db = require("../models");
+
+module.exports = function(app){
+
+
+    // POST new user
+    app.post("/api/users", function(req, res){
+        db.Products.create(req.body).then(function(dbUser){
+            res.json(dbUser)
```
Comment (ALMaclaine) [inline routes/login-routes.js]: Prefer arrow functions for short functions without this.
.then((dbUser) => res.json(dbUser)

File: routes/login-routes.js [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+        });
+    });
+
+    // DELETE user
+    app.delete("/api/user/:id", function(req, res) {
+        db.Products.destroy({
+        where: {
+            id: req.params.id
+        }
+        }).then(function(dbUser) {
```
Comment (ALMaclaine) [inline routes/login-routes.js]: Prefer arrow functions for short functions without this.
.then((dbUser) => res.json(dbUser)

File: routes/login-routes.js [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+
+    // UPDATE user info
+    app.put("/api/user", function(req, res) {
+        db.Products.update(
+        req.body,
+        {
+            where: {
+            id: req.body.id
+            }
+        }).then(function(dbUser) {
```
Comment (ALMaclaine) [inline routes/login-routes.js]: Prefer arrow functions for short functions without this.
.then((dbUser) => res.json(dbUser)

**Review by ALMaclaine — COMMENTED** [review]
Give all non-generated files sensible names, for example your images.

Also I didn't mark every instance of repeated comments. In general look for everywhere you can protect from req.body api changes, prefer arrow functions for very short functions especially when "this" is not involved and prefer es6 object intializer syntax over explicit property naming.

File: models/login.js [hunk 20]
```diff
@@ -0,0 +1,36 @@
+module.exports = function(sequelize, DataTypes) {
+    const User = sequelize.define("Products", {
```
[the remark this answers is not in the document]
Comment (nmclear) [reply models/login.js]: Thank you, I missed that error. It should be User and now functions properly.

File: config/config.js (hunk 3 above)
Comment (nmclear) [reply config/config.js]: I switched over to a config.js to allow for an env password, but forgot to delete the config.json. Thanks for catching it!

File: public/assets/js/inventory-public.js [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+
+    
+    // Check for seller value
+    function handleSellerInventory(event){
+        event.preventDefault();
+
+        // val should be seller id (number)
+        const sellerId = $('.sellerInventory').val().trim();
+        
+        if(!NaN(sellerId)){
```
[the remark this answers is not in the document]
Comment (nmclear) [reply public/assets/js/inventory-public.js]: You are correct, NaN is not a function. I meant to have isNaN()

File: routes/inventory-routes.js [hunk 22]
```diff
@@ -0,0 +1,85 @@
+const db = require("../models");
+
+module.exports = function(app){
+
+    // GET full inventory
+    app.get("/api/inventory", function(req,res){
+
+        db.Inventory.findAll({
+            include: [db.Products][db.Seller]
```
[the remark this answers is not in the document]
Comment (nmclear) [reply routes/inventory-routes.js]: [db.Products][db.Seller] does not work. I switched to Taylor's recommendation of [db.Products, db.Seller] and it works. However, I may not need this line of code anymore.

File: routes/inventory-routes.js (hunk 2 above)
Comment (nmclear) [reply routes/inventory-routes.js]: [db.Products][db.Seller] does not work. I switched to your recommendation of [db.Products, db.Seller] and it works. However, I may not need this line of code anymore. I added include: [db.Products] to the seller-routes. Now when I load a seller, it list all the products they are selling plus stock.

File: routes/login-routes.js (hunk 17 above)
Comment (nmclear) [reply routes/login-routes.js]: Thank you, I'll go through and switch to ES6 arrow functions where appropriate.

**Pull request closed without merging**